website/docs/r/servicebus_namespace_network_rule_set.html.markdown

```diff
@@ -0,0 +1,111 @@
+---
+subcategory: "Messaging"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_servicebus_namespace_network_rule_set"
+description: |-
+  Manages a ServiceBus Namespace Network Rule Set Set.
+---
+
+# azurerm_servicebus_namespace_network_rule_set
+
+Manages a ServiceBus Namespace Network Rule Set Set.
+
+## Example Usage
+
+```hcl
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_servicebus_namespace" "example" {
+  name                = "example-sb-namespace"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  sku                 = "Premium"
+
+  capacity = 1
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "example-vnet"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  address_space       = ["172.17.0.0/16"]
+  dns_servers         = ["10.0.0.4", "10.0.0.5"]
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "default"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefix       = "172.17.0.0/24"
+
+  service_endpoints = ["Microsoft.ServiceBus"]
+}
+
+resource "azurerm_servicebus_namespace_network_rule_set" "example" {
+  namespace_name      = azurerm_servicebus_namespace.example.name
+  resource_group_name = azurerm_resource_group.example.name
+
+  default_action = "Deny"
+
+  network_rules {
+    subnet_id                            = azurerm_subnet.example.id
+    ignore_missing_vnet_service_endpoint = false
+  }
+
+  ip_rules = ["1.1.1.1"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `resource_group_name` - (Required) Specifies the name of the Resource Group where the ServiceBus Namespace Network Rule Set should exist. Changing this forces a new resource to be created.
+
+* `namespace_name` - (Required) Specifies the ServiceBus Namespace name to which to attach the ServiceBus Namespace Network Rule Set. Changing this forces a new resource to be created.
+
+~> **NOTE:** The ServiceBus Namespace must be `Premium` in order to attach a ServiceBus Namespace Network Rule Set.
+
+* `default_action` - (Optional) Specifies the default action for the ServiceBus Namespace Network Rule Set. Possible values are `Allow` and `Deny`. Defaults to `Deny`.
+
+* `ip_rules` - (Optional) A list of IP filter masks that are added to allow access to the ServiceBus Namespace.
```

Comment:
Based on the `network_acls` of `azurerm_key_vault` and the assumption CIDR Blocks work as well:
```suggestion
* `ip_rules` - (Optional) One or more IP Addresses, or CIDR Blocks which should be able to access the ServiceBus Namespace.
```

Reply:
Done